src/App.jsx

```diff
@@ -196,15 +246,22 @@ function App() {
       body: JSON.stringify(payload),
     };
     try {
+      /* Before reducer
       setIsSaving(true);
-      const resp = await fetch(encodeUrl(sortField, sortDirection, queryString ), options);
+      */
+      dispatch({ type: todoActions.startRequest });
+      const resp = await fetch(encodeUrl(sortField, sortDirection, queryString), options);
       if (!resp.ok) {
         throw new Error(resp.message);
       }
     } catch (error) {
+      /* Before reducer
       console.error(error);
       setErrorMessage(`${error.message}. Reverting todo...`);
+      */
+      dispatch({ type: todoActions.setLoadError, error: error });
```

Comment:
Future implementation: Having two dispatch function for failed requests is a great approach. However, since two async action will always happen on a failed request this logic can be further optimized into one action.